pkg/image/api/sort.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"sort"
+
+	"k8s.io/kubernetes/pkg/util"
+)
+
+type tag struct {
+	Name    string
+	Created util.Time
+}
+
+type byCreationTimestamp []tag
+
+func (t byCreationTimestamp) Len() int {
+	return len(t)
+}
+
+func (t byCreationTimestamp) Less(i, j int) bool {
+	return t[i].Created.Time.After(t[j].Created.Time)
+}
+
+func (t byCreationTimestamp) Swap(i, j int) {
+	t[i], t[j] = t[j], t[i]
+}
+
+// SortStatusTags sorts the status tags of an image stream based on
+// the latest created
+func SortStatusTags(tags map[string]TagEventList) []string {
```

Comment:
Discussed on IRC, I'm fine with current impl. If needed we'll change it into more generic solution.